webapp/packages/plugin-connections-administration/src/Administration/Connections/CreateConnectionService.ts

```diff
@@ -109,26 +104,31 @@ export class CreateConnectionService {
     this.activateMethod(defaultId);
   }
 
-  setConnectionTemplate(projectId: string, config: ConnectionConfig, availableDrivers: string[]): void {
-    this.data = new ConnectionFormState(
-      this.projectsService,
-      this.projectInfoResource,
-      this.connectionFormService,
-      this.connectionInfoResource,
-      this.connectionInfoOriginResource,
-    );
-
-    this.data.closeTask.addHandler(this.cancelCreate.bind(this));
+  private get optionsPart() {
+    return this.data ? getConnectionFormOptionsPart(this.data) : null;
+  }
 
-    this.data
-      .setOptions('create', 'admin')
-      .setConfig(projectId, config)
-      .setAvailableDrivers(availableDrivers || []);
+  setConnectionTemplate(projectId: string, config: ConnectionConfig, availableDrivers: string[]) {
+    this.clearConnectionTemplate();
+    this.data = new ConnectionFormState(this.serviceProvider, this.connectionFormService, {
+      projectId,
+      availableDrivers,
+      submitType: null,
+      type: 'admin',
+      requiredNetworkHandlersIds: [],
+    });
 
-    this.data.load();
+    runInAction(() => {
+      this.optionsPart!.state = {
+        ...this.optionsPart!.state,
+        ...config,
+      };
```

Comment:
in mobx context `Object.assign(this.optionsPart!.state, config)` is more preferable